### PR TITLE
More cuda test fixes

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Fixes roll and scatter_nd cuda kernels
(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.konduit.ai/multi-project/how-to-guides/contribute/eclipse-contributors) page for details
- [X ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ X] Created tests for any significant new code additions.
- [X ] Relevant tests for your changes are passing.

### CODE REVIEW

File: libnd4j/include/ops/declarable/generic/parity_ops/roll.cpp [hunk 1]
```diff
@@ -107,6 +107,7 @@ CONFIGURABLE_OP_IMPL(roll, 1, 1, true, 0, 0) {
     return sd::Status::OK;
   }
 
+  sd_printf("Shift is linear %d\n",shiftIsLinear);
```
Comment (treo) [inline libnd4j/include/ops/declarable/generic/parity_ops/roll.cpp]: left over from debugging?

File: libnd4j/include/ops/declarable/generic/transforms/scatter_add.cpp [hunk 2]
```diff
@@ -35,6 +35,7 @@ OP_IMPL(scatter_add, 3, 1, true) {
     auto indices = INPUT_VARIABLE(1);
     auto updates = INPUT_VARIABLE(2);
 
+    sd_printf("Scatter add base op\n",0);
```
Comment (treo) [inline libnd4j/include/ops/declarable/generic/transforms/scatter_add.cpp]: left over from debugging?

File: libnd4j/include/ops/declarable/helpers/cuda/roll.cu [hunk 3]
```diff
@@ -252,40 +252,17 @@ static void rollFunctorFull_(NDArray *input, NDArray *output, std::vector<sd::Lo
...
-          rollKernelFullAnyDimensionStage1<T><<<1, 256, 1024, *(output->getContext()->getCudaStream())>>>(
-              output->specialBuffer(), packZ->platformShapeInfo(), packZ->platformOffsets(), output->specialBuffer(),
-              packZ->platformShapeInfo(), packZ->platformOffsets(), numTads, tadLength, dim, sizeAt, theShift);
-
-          rollKernelFullAnyDimensionStage2<T><<<1, 256, 1024, *(output->getContext()->getCudaStream())>>>(
-              output->specialBuffer(), packZ->platformShapeInfo(), packZ->platformOffsets(), output->specialBuffer(),
-              packZ->platformShapeInfo(), packZ->platformOffsets(), numTads, tadLength, dim, sizeAt, theShift);
-        }
-      }
+    sd_printf("Last dimension case\n",0);
```
Comment (treo) [inline libnd4j/include/ops/declarable/helpers/cuda/roll.cu]: left over from debugging?

File: libnd4j/include/ops/declarable/helpers/cuda/scatter.cu [hunk 4]
```diff
@@ -491,39 +489,36 @@ SD_KERNEL static void scatterNDCuda(const int opCode, const void *vx, const sd::
   }
   __syncthreads();
 
+  printf("is1Dcase %d xRank %d yRank %d zRank %d yLen %d\n",is1Dcase, xRank, yRank, zRank,yLen);
```
Comment (treo) [inline libnd4j/include/ops/declarable/helpers/cuda/scatter.cu]: left over from debugging?

File: libnd4j/include/ops/declarable/helpers/cuda/scatter.cu [hunk 5]
```diff
@@ -491,39 +489,36 @@ SD_KERNEL static void scatterNDCuda(const int opCode, const void *vx, const sd::
...
+    for (sd::LongType j = 0; j < xLastDim; ++j) {  // first xRank-1 coordinates in yCoords are the same for y and x
+      yCoords[xRank - 1] = j;
+      zCoords[j] = x[shape::getOffset(xShapeInfo, yCoords)];
     }
 
+    for (sd::LongType j = xLastDim + 1; j < zRank; ++j) zCoords[j] = yCoords[yRank - zRank + j];
+
+    zOffset = shape::getOffset(zShapeInfo, zCoords);
+
+    printf("zOffset %d yOffset %d value at zOffset %f value at yOffset %f \n",zOffset,yOffset,z[zOffset],y[yOffset]);
```
Comment (treo) [inline libnd4j/include/ops/declarable/helpers/cuda/scatter.cu]: left over from debugging?

File: libnd4j/include/ops/declarable/helpers/cuda/scatter.cu [hunk 6]
```diff
@@ -491,39 +489,36 @@ SD_KERNEL static void scatterNDCuda(const int opCode, const void *vx, const sd::
...
     switch (opCode) {
       case pairwise::Add:
         z[zOffset] += y[yOffset];
+        printf("New zOffset %d yOffset %d value at zOffset %f value at yOffset %f \n",zOffset,yOffset,z[zOffset],y[yOffset]);
```
Comment (treo) [inline libnd4j/include/ops/declarable/helpers/cuda/scatter.cu]: left over from debugging?

**Review by treo — CHANGES_REQUESTED** [review]
Looks like you've left a few printfs from debugging.

**Comment by agibsonccc** [comment]
@treo  resolved

**Review by treo — APPROVED** [review]

**Pull request merged**